roles/step_ca/meta/argument_specs.yml

```diff
@@ -47,6 +47,23 @@ argument_specs:
         default: no
         description: Create keys to sign SSH certificates
       # Existing cert options
+      step_ca_existing_root:
+        type: str
+        choices:
+          - remote
+          - local
+        default: remote
```

Comment:
Actually setting to remote and not passing `existing_*` will just ignore this option. I.e. we only need 2 choices

Reply:
Ah I missed that, good call! That said, I think I prefer a third, explicit option to avoid potential confusion. I've already made that adjustment with my last commit, I'll add tests and then merge in the coming days

Reply:
Ok, just 2 things, we should add a check that if it is false, but existing is passed then it should fail because the intent is ambiguous both in if they want new root and where existing would be located. And second, doesn't `false` there get converted to string `"false"` which would have resolve to true in the checks? In the parameter file, it is specified as string

Reply:
> Ok, just 2 things, we should add a check that if it is false, but existing is passed then it should fail because the intent is ambiguous

Correct! I've already added that [here](https://github.com/maxhoesel-ansible/ansible-collection-smallstep/blob/a688323d01f2bd7493b74f9ff36cba9de378d93f/roles/step_ca/tasks/check.yml#L15-L19)

> doesn't false there get converted to string "false"

Honestly? I'm not sure, I'll have to check that later when I get around to writing the tests. If it does, I'll make the required modifications. That said, I'm pretty sure that the `argument_spec` dict is only used for validation and doesn't actually cast anything. Could be wrong on that though.